### ISSUE TITLE
.rubocop_todo.yml mentions file explicitly Excluded in .rubocop.yml

### ISSUE THREAD

**Issue by jcraigk** [rubocop-hq/rubocop#8857]
## Expected behavior

When running `rubocop --auto-gen-config --auto-gen-only-exclude --exclude-limit 100` on a project where the following is present in `.rubocop.yml`:
```
Metrics/MethodLength:
  Exclude:
    - path/to/excluded/file.rb
```
I expect that the `path/to/excluded/file.rb` will not appear in `.rubocop_todo.yml` under the `Metrics/MethodLength` cop since I have already explicitly asked Rubocop to Exclude it. I know the Exclude is working because when I run `rubocop` by itself, no offenses are reported on that file even though there are offenses present within it.

## Actual behavior

The following appears in `.rubocop_todo.yml`:

```
# Offense count: 6
# Configuration parameters: CountComments, Max, CountAsOne, ExcludedMethods.
Metrics/MethodLength:
  Exclude:
    - 'path/to/included/file.rb'
    - 'path/to/excluded/file.rb'
```

Note that the `path/to/included/file.rb` contains violations of the `Metrics/MethodLength` rule. While that is also true of `path/to/excluded/file.rb`, we explicitly Exclude that file in our `.rubocop.yml` for that specific cop.

Note that if we delete `path/to/included/file.rb` and run `rubocop --auto-gen-config --auto-gen-only-exclude --exclude-limit 100` again, neither of the files appears as offenders in the `.rubocop_todo.yml`.

## Steps to reproduce the problem

1. Create two local Ruby files, both containing violations of `Metrics/MethodLength`
2. Create a `.rubocop.yml` file that contains a rule to Exclude only one of these files for that specific cop:
```
Metrics/MethodLength:
  Exclude:
    - path/to/excluded/file.rb
```
3. Run `rubocop --auto-gen-config --auto-gen-only-exclude --exclude-limit 100`
4. Observe that the `path/to/excluded/file.rb` appears under `Metrics/MethodLength` in `.rubocop_todo.yml`

## RuboCop version

```
0.92.0 (using Parser 2.7.2.0, rubocop-ast 0.7.1, running on ruby 2.6.6 x86_64-darwin19)
```

**Comment by jcraigk** [comment]
Quick followup: I may have misunderstood how the `.rubocop_todo.yml` is generated for each rule. It appears that Rubocop can't handle merging rules from `.rubocop.yml` and `.rubocop_todo.yml` - an `Exclude` for a specific cop in `.rubocop.yml` gets completely replaced with the contents of the same `Exclude` in `.rubocop_todo.yml` which perhaps makes the observed behavior necessary - the `.rubocop_todo.yml` has to replicate every `Exclude` found in `.rubocop.yml` it seems. Probably a heavy lift, but it would be great if Rubocop were capable of merging the rules rather than replacing them outright for each cop.

**Comment by jonas054** [comment]
You're right. RuboCop copies `Exclude` properties from existing configuration into `.rubocop_todo.yml`. For `Excludes` in default configuration this works well. It's the configuration in `.rubocop_todo.yml` that will take precedence. For `Excludes` in custom configuration (`.rubocop.yml`) however, it will not work as intended. The `Exclude`s in `.rubocop.yml` will override the same properties in `.rubocop_todo.yml` since the inheriting configuration does this.

If you add to your `.rubocop.yml` this setting:
```yaml
inherit_mode:
  merge:
    - Exclude
```
the situation will be better. The `Exclude`s in `.rubocop.yml` will not override the ones in `.rubocop_todo.yml`. The only remaining problem is that some `Exclude` items are copied unnecessarily into `.rubocop_todo.yml`.

This behavior from `rubocop --auto-gen-config` is incorrect IMO, and should be fixed. I propose the following changes:

1. If `inherit_mode` is `merge` for `Exclude`, items that are excluded anyway through merge should not be copied into `.rubocop_todo.yml`.
2. If `inherit_mode` is NOT `merge` for `Exclude`, a warning should be output to indicate that the user needs to set `inherit_mode` or remove `Exclude` properties in `.rubocop.yml`.

An alternative to (2) would be to generate `inherit_mode` into `.rubocop.yml`, but that's a bit risky IMO.

I have a PR in the works.
